src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1298,13 +1300,22 @@ class ObjectGenerator : public Generator<bsoncxx::document::value> {
           _keyGen{stringGenerator(node["havingKeys"], generatorArgs)},
           _valueGen{
               valueGenerator<false, UniqueAppendable>(node["andValues"], generatorArgs, parsers)},
-          _nTimesGen{intGenerator(extract(node, "withNEntries", "^Object"), generatorArgs)} {}
+          _nTimesGen{intGenerator(extract(node, "withNEntries", "^Object"), generatorArgs)},
+          _allowDuplicateKeys{node["allowDuplicateKeys"].to<bool>()} {}
 
     bsoncxx::document::value evaluate() override {
         bsoncxx::builder::basic::document builder;
         auto times = _nTimesGen->evaluate();
+
+        std::unordered_set<std::string> usedKeys;
         for (int i = 0; i < times; ++i) {
-            _valueGen->append(_keyGen->evaluate(), builder);
+            auto key = _keyGen->evaluate();
+            if (_allowDuplicateKeys || usedKeys.insert(key).second) {
+                _valueGen->append(key, builder);
+            }
+        }
+        if (!_allowDuplicateKeys && usedKeys.size() < times) {
```

Comment:
This is probably too verbose: reduce severity or not log at all?

Reply:
I'm torn here. I think we can just leave as-is to see how commonly this actually comes up. This (most likely) points to incorrect configuration if happens often, so spamming the logs might be the best canary we can shove into this coal-mine.

Reply:
For that case it might be easier to just have `{a: {^Choose}, b: {^Choose}, c:{^Choose}}` since you know you want an object with exactly 3 keys (a,b,c).

Reply:
I actually wanted and object which sometimes misses one or two of the keys